csharp/src/AddressBook/Addressbook.cs

```diff
@@ -241,7 +236,7 @@ public sealed partial class Person : pb::IMessage<Person> {
             break;
           }
           case 34: {
-            input.ReadMessageArray(phone_, global::Google.Protobuf.Examples.AddressBook.Person.Types.PhoneNumber.Parser);
+            phone_.AddEntriesFrom(input, _repeated_phone_codec);
```

Comment:
I wondered about that, but note that the codec is static whereas the list is a per-instance field. Making it part of the list would mean having the same value in every list, and potentially creating a new codec each time.